### ISSUE TITLE
RET504 off-by-1  line / bogus error when returning an argument

### ISSUE THREAD

**Issue by hynek** [charliermarsh/ruff#3992]
Consider the following file `t.py`:

```python
def str_to_bool(val):
    if isinstance(val, bool):
        return val
    val = val.strip().lower()
    if val in ("1", "true", "yes"):
        return True

    return False
```

Running `pipx run ruff check --isolated --select RET504 t.py` gives me:

```
t.py:3:16: RET504 Unnecessary variable assignment before `return` statement
```

So it points at the `return val` that hasn't assigned anything.

If I set val by hand instead of passing it in as an argument, there is no error.

---

```console
$ pipx run ruff --version
ruff 0.0.261
```

no `pyproject.toml`

**Comment by charliermarsh** [comment]
👍 Definitely a bug.
